docs/tutorials/nodes-tutorial.ipynb

```diff
@@ -0,0 +1,376 @@
+    "Nodes in the graph have to be constructed from callables.\n",
+    "When a view requests data from a parent node, the callable is called.\n",
+    "Typically, the callables will be a function that take in a data array as an input,\n",
+    "and returns a data array as an output.\n",
+    "\n",
+    "Keeping your inputs and outputs as data arrays is useful because figure views will only accept data arrays as input.\n",
+    "That said, nodes that produce other outputs are very common, for example when using interactive widgets.\n",
+    "\n",
+    "In the small example above, the node at the top of the graph has no parents,\n",
+    "and its callable is simply a `lambda` function with no arguments that just returns the input data.\n",
+    "\n",
+    "## Constructing the graph\n",
+    "\n",
+    "To make our graph above, we begin by generating some 2D data,\n",
+    "and feeding that to an `input_node`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a0db9e7-ffc2-443b-83eb-7c8904247062",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib widget\n",
+    "import plopp as pp\n",
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "\n",
+    "da = pp.data.dense_data_array(ndim=2)\n",
+    "noise = sc.array(dims=da.dims, values=np.random.random(da.shape), unit=da.unit)\n",
+    "da.data = (da.data * 5.0) + noise\n",
+    "da"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "047dba7c-a766-4222-bfe6-81c91f57f77d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = pp.input_node(da)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd0a32c9-6019-4734-b0ce-ec15db489703",
+   "metadata": {},
+   "source": [
+    "Next, we give that input node as input to a two-dimensional figure:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a11ed71-c2c1-4fb7-9f7b-0841dd506f62",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = pp.figure2d(a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6864bfb0-66c7-4425-8ed4-1349beddb900",
+   "metadata": {},
+   "source": [
+    "We can display our graph using the `show_graph` function,\n",
+    "which will accept as input any node in the graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f74ed773-2903-4699-8d6c-f9f848a17e6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pp.show_graph(a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96fb3b9e-4e62-48fa-b697-a22aa72ffc8c",
+   "metadata": {},
+   "source": [
+    "As shown in the graph above, each node has a callable (in this case a `lambda` function) and an `id`.\n",
+    "\n",
+    "The figure can directly be displayed in the notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e6365c1-68cf-4c92-8903-726a85aff451",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig"
```

Comment:
The figures look a bit blurry. Is it possible / feasible to use PDFs instead of PNGs?

Reply:
Unfortunately not :-(

Reply:
I can try to fiddle with the dpi